pkg/services/k8s/apiserver/authorization/grafana_authorizer.go

```diff
@@ -0,0 +1,72 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+	"github.com/grafana/grafana/pkg/services/org"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+type GrafanaAuthorizer struct {
+}
+
+func errorMessageForGrafanaOrgRole(grafanaOrgRole string, a authorizer.Attributes) string {
+	return fmt.Sprintf("Grafana org role (%s) didn't allow %s access on requested resource=%s, path=%s", grafanaOrgRole, a.GetVerb(), a.GetResource(), a.GetPath())
+}
+
+func errorMessageForK8sGroup(k8sGroups []string, a authorizer.Attributes) string {
+	return fmt.Sprintf("K8s groups (%v) didn't allow %s access on requested resource=%s, path=%s", k8sGroups, a.GetVerb(), a.GetResource(), a.GetPath())
+}
+
+func NewGrafanaAuthorizer() *GrafanaAuthorizer {
+	return &GrafanaAuthorizer{}
+}
+
+func (auth GrafanaAuthorizer) Authorize(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+	userInfo, ok := request.UserFrom(ctx);
+	if !ok {
+		return authorizer.DecisionDeny, fmt.Sprintf("could not determine user info from context, denied %s access on requested resource=%s, path=%s", a.GetVerb(), a.GetResource(), a.GetPath()), nil
+	}
+	extra := userInfo.GetExtra()
+	groups := userInfo.GetGroups()
+
+	// TODO: take orgId into account in relation to namespaced resources
+	// if orgId := extra["org-id"]; len(orgId) > 0 {}
+	
+	// We are being called using a Grafana token
+	if orgRole := extra["org-role"]; len(orgRole) > 0 {
+		switch org.RoleType(orgRole[0]) {
```

Comment:
can there be multiple roles?
```suggestion
        // TODO: handle multiple roles
		switch org.RoleType(orgRole[0]) {
```

Reply:
In kubernetes, there can be multiple roles in that header, but in [Grafana](https://github.com/grafana/grafana/blob/k8s-grafana-apiserver/pkg/services/k8s/apiserver/service.go#L223), we are only setting the header to a single value.

From the looks of it, signedInUser can only have 1 org role.